### PR TITLE
IDENTITY-6090: Option to Skip role validation in JDBC scope validator handler

### PR DESCRIPTION
**Pull request by madumalt**
https://wso2.org/jira/browse/IDENTITY-6090
@isharak @IsuraD

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java [hunk 1]
```diff
@@ -53,6 +54,12 @@
  */
 public class JDBCScopeValidator extends OAuth2ScopeValidator {
 
+    // The following constants are as same as the constants defined in
+    // org.wso2.carbon.apimgt.keymgt.handlers.ResourceConstants.
+    // If any changes are taking place in that these should also be updated accordingly.
+    public static final String CHECK_TRUE = "true";
```
Comment (lmanohara) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java]: change the access modifiers to private.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java [hunk 2]
```diff
@@ -123,6 +130,14 @@ public boolean validateScope(AccessTokenDO accessTokenDO, String resource) throw
             return false;
         }
 
+        // If GrantType is SAML20_BEAERER and SAML2_ASSERTION_ENABLED system property is set to true
+        // Avoid validating user roles
+        String isSAML2Enabled = System.getProperty(SAML2_ASSERTION_ENABLED);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java]: Where does this System property get set?
Can there be a case where two different entities set this property concurrently that will cause unexpected behaviours?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java (hunk 1 above)
Comment (madumalt) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java]: done

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java [hunk 3]
```diff
@@ -123,6 +130,14 @@ public boolean validateScope(AccessTokenDO accessTokenDO, String resource) throw
...
+        if (GrantType.SAML20_BEARER.toString().equalsIgnoreCase(accessTokenDO.getGrantType())
+                && isSAML2Enabled != null && CHECK_TRUE.equalsIgnoreCase(isSAML2Enabled)) {
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java]: Why we need a constant to check true?

Cann't we just use `Boolean.parseBoolean(isSAML2Enabled)`?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java (hunk 2 above)
Comment (madumalt) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java]: this property is set at server start. Thus wll not have such cases.

**Pull request closed without merging**